main.tf

```diff
@@ -1,32 +1,135 @@
 data "tencentcloud_instance_types" "default" {
-  cpu_core_count = var.cpu_core_count
-  gpu_core_count = var.gpu_core_count
-  memory_size    = var.memory_size
+  cpu_core_count   = var.cpu_core_count
+  gpu_core_count   = var.gpu_core_count
+  memory_size      = var.memory_size
+  exclude_sold_out = true
+}
+
+data "tencentcloud_vpc_route_tables" "default" {
+  vpc_id           = var.vpc_id
+  association_main = true
+}
+
+locals {
+  create_vpc         = var.create_vpc
+  custom_route_table = var.create_route_table == false || local.create_vpc ? false : length(var.destination_cidrs) > 0
+  availability_zones = length(var.availability_zones) > 0 ? var.availability_zones : data.tencentcloud_instance_types.default.instance_types.*.availability_zone
+  route_table_id = var.route_table_id != "" ? var.route_table_id : local.custom_route_table ? tencentcloud_route_table.route_table[0].id : local.create_vpc ? tencentcloud_vpc.vpc[0].default_route_table_id : try(
+    data.tencentcloud_vpc_route_tables.default.instance_list[0].route_table_id,
+    null
+  )
 }
 
 resource "tencentcloud_vpc" "vpc" {
-  count        = var.vpc_id == "" ? 1 : 0
+  count = local.create_vpc ? 1 : 0
+  lifecycle {
+    precondition {
```

Comment:
You've specify `precondition` block which required Terraform version v1.2.0 and later, but required version in `versions.tf` was `required_version = ">= 0.12"`